src/main/java/org/prebid/server/privacy/gdpr/vendorlist/VendorListService.java

```diff
@@ -182,17 +198,17 @@ private void fetchNewVendorListFor(int version) {
         final String url = endpointTemplate.replace(VERSION_PLACEHOLDER, String.valueOf(version));
 
         httpClient.get(url, defaultTimeoutMs)
-                .compose(response -> processResponse(response, version))
-                .compose(vendorListResult -> saveToFileAndUpdateCache(vendorListResult, version))
-                .recover(exception -> failResponse(exception, version));
+                .map(response -> processResponse(response, version))
+                .recover(exception -> failResponse(exception, version))
+                .compose(vendorListResult -> saveToFileAndUpdateCache(vendorListResult, version));
```

Comment:
It's strange that `saveToFileAndUpdateCache` does not show the result of the file saving/
Doesn't `recover` method  in the end of pipeline is  safer ?

Reply:
Good question. From the logic point of view we don't need to cover `filesystem` errors (all we need is already implemented in `fileSystem.writeFile(...)` call) but maybe you're right and i need to make it in `clean-code` style.